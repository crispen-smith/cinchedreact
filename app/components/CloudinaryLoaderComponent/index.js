/**
 *
 * CloudinaryLoaderComponent
 *
 */

import React from 'react';
import Dropzone from 'react-dropzone';
import request from 'superagent';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ErrorBar from '../ErrorBar';

const CLC = styled.div`
  border-radius: 10px;

  input {
    border-radius: 20px;
  }
`;

export function onImageDrop(files) {
  const upload = request
    .post(this.CLOUDINARY_UPLOAD_URL)
    .field('upload_preset', this.props.preset)
    .field('file', files[0]);

  upload.end((err, response) => {
    if (err) {
      this.setState({ error: true, err });
    }
    if (response.body.secure_url !== '') {
      const action = this.props.completionHandler(response);
      this.props.dispatch(action);
    }
  });
}

/* eslint-disable react/prefer-stateless-function */
class CloudinaryLoaderComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = { error: false };
    this.CLOUDINARY_UPLOAD_URL =
      'https://api.cloudinary.com/v1_1/www-cinchedtight-com/upload';
    this.onImageDrop = onImageDrop.bind(this);
  }

  render() {
    const error = this.state.error ? (
      <ErrorBar message={this.state.err} />
    ) : null;
    return (
      <CLC>
        <Dropzone multiple={false} accept="image/*" onDrop={this.onImageDrop}>
          <p>{this.props.dropMessage}</p>
          {error}
        </Dropzone>
      </CLC>
    );
  }
}
CloudinaryLoaderComponent.propTypes = {
  completionHandler: PropTypes.func.isRequired, // eslint-disable-line react/no-unused-prop-types
  preset: PropTypes.oneOf(['corset', 'lingerie', 'latex', 'client']), // eslint-disable-line react/no-unused-prop-types
  dropMessage: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired, // eslint-disable-line react/no-unused-prop-types
};

export default CloudinaryLoaderComponent;
