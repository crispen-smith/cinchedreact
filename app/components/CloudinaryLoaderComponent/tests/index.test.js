import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import CloudinaryLoaderComponent from '../index';

describe('<CloudinaryLoaderComponent />', () => {
  it('Renders a CloudinaryLoaderComponent without an error bar', () => {
    const dispatch = func => func();
    const completionHandler = () => null;
    const preset = 'corset';
    const dropMessage = 'Dropped';
    const wrapper = renderer
      .create(
        <CloudinaryLoaderComponent
          dispatch={dispatch}
          completionHandler={completionHandler}
          preset={preset}
          dropMessage={dropMessage}
        >
          TEST
        </CloudinaryLoaderComponent>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('Renders a CloudinaryLoaderComponent with an error bar', () => {
    const dispatch = func => func();
    const completionHandler = () => null;
    const preset = 'corset';
    const dropMessage = 'Dropped';
    const err = 'Error Message';

    const CLC = mount(
      <CloudinaryLoaderComponent
        dispatch={dispatch}
        completionHandler={completionHandler}
        preset={preset}
        dropMessage={dropMessage}
      />,
    );
    CLC.setState({ error: true, err }, () => {
      expect(CLC.find('ErrorBar')).toHaveLength(1);
    });
  });
});
