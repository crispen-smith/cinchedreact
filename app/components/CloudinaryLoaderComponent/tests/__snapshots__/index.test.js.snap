// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloudinaryLoaderComponent /> Renders a CloudinaryLoaderComponent 1`] = `
<div
  className="CloudinaryLoaderComponent__CLC-kyCANm dMYuko"
>
  <div
    aria-disabled={false}
    className=""
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "borderColor": "#666",
        "borderRadius": 5,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "height": 200,
        "position": "relative",
        "width": 200,
      }
    }
  >
    <p>
      Dropped
    </p>
    <input
      accept="image/*"
      autoComplete="off"
      disabled={false}
      multiple={false}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
</div>
`;

exports[`<CloudinaryLoaderComponent /> Renders a CloudinaryLoaderComponent without an error bar 1`] = `
<div
  className="CloudinaryLoaderComponent__CLC-kyCANm dMYuko"
>
  <div
    aria-disabled={false}
    className=""
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    style={
      Object {
        "borderColor": "#666",
        "borderRadius": 5,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "height": 200,
        "position": "relative",
        "width": 200,
      }
    }
  >
    <p>
      Dropped
    </p>
    <input
      accept="image/*"
      autoComplete="off"
      disabled={false}
      multiple={false}
      onChange={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0.00001,
          "pointerEvents": "none",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      type="file"
    />
  </div>
</div>
`;
